scripts/compress_jointcal_cfht_test_data.py

```diff
@@ -1,41 +1,75 @@
+contain the files that are handled by this script; refcats, export files, etc.
+should live elsewhere.
 
 No need to re-compress with fpack, as they are all already CompImageHDUs.
 Requires that validation_data_cfht be setup, that this be run from
 `testdata_jointcal/`.
 """
 from astropy.io import fits
 import os
-import shutil
 import glob
-import subprocess
+import shutil
+import sys
+
+import pyarrow.parquet
 
 import lsst.utils
 
-inpath = lsst.utils.getPackageDir("validation_data_cfht")
+# We only use these detectors for the jointcal tests.
+detectors = (12, 13, 14, 21, 22, 23)
+
+inpath = os.path.join(lsst.utils.getPackageDir("validation_data_cfht"), "repo")
 
 # cleanup directories before copying new data
-shutil.rmtree('cfht/calexp', ignore_errors=True)
-shutil.rmtree('cfht/config', ignore_errors=True)
-shutil.rmtree('cfht/src', ignore_errors=True)
-shutil.rmtree('cfht/metadata', ignore_errors=True)
-shutil.rmtree('cfht/schema', ignore_errors=True)
+shutil.rmtree('cfht/repo', ignore_errors=True)
 
 # copy new data
-shutil.copytree(f"{inpath}/data/output/calexp/", "cfht/calexp/")
-shutil.copytree(f"{inpath}/data/output/config/", "cfht/config/")
-shutil.copytree(f"{inpath}/data/output/src/", "cfht/src/")
-shutil.copytree(f"{inpath}/data/output/schema/", "cfht/schema/")
+paths = glob.glob(f"{inpath}/singleFrame/*")
+if len(paths) > 1:
+    print(f"ERROR: Found {len(paths)} singleFrame output paths.")
+    print("ERROR: Only run this script on a single-pipeline-run validation_data_cfht output repo.")
+    sys.exit(-1)
+run = paths[0].split('/')[-1]
+shutil.copytree(f"{inpath}/singleFrame/{run}/calexp/",
+                f"cfht/repo/singleFrame/{run}/calexp")
+# TODO: do we need the config files?
+# shutil.copytree(f"{inpath}/data/output/config/", "cfht/repo/config/")
+shutil.copytree(f"{inpath}/singleFrame/{run}/sourceTable_visit/",
+                f"cfht/repo/singleFrame/{run}/sourceTable_visit/")
+shutil.copytree(f"{inpath}/singleFrame/{run}/visitSummary/",
+                f"cfht/repo/singleFrame/{run}/visitSummary/")
+shutil.copytree(f"{inpath}/skymaps", "cfht/repo/skymaps")
+shutil.copytree(f"{inpath}/MegaPrime/calib", "cfht/repo/MegaPrime/calib")
+
+
+def remove_detector_files(path):
+    """Remove output files for detectors we don't need."""
+    for file in glob.glob(path):
+        ccd = int(file.split('ccd')[-1].split('_')[0])
+        if ccd not in detectors:
+            os.remove(file)
+
+
+remove_detector_files(f"cfht/repo/singleFrame/{run}/calexp/*/r/r.MP9601/*/calexp*.fits")
+
+
+def remove_detector_parquet(path):
+    """Remove rows from parquet files for detectors we don't need."""
+    for filename in glob.glob(path):
+        data = pyarrow.parquet.ParquetFile(filename).read(use_pandas_metadata=True).to_pandas()
+        match = data['detector'].isin(detectors)
+        data[match].to_parquet(filename)
+
+
+remove_detector_parquet(f"cfht/repo/singleFrame/{run}/sourceTable_visit/*/r/r.MP9601/*/*.parq")
```

Comment:
It turns out that doing it this way was good, because it pointed out a bug in jointcal! Jointcal's gen3 data loading couldn't handle missing detectors, which is a thing we may have in the future in real data (with the "don't save catalogs for failed detector processing" thing).

Fix is here:
https://github.com/lsst/jointcal/pull/205/commits/b2d80f4cdb0f889464d3a4e8652de1fc2dfee8be

Reply:
So the problem was when there is a mismatch between what's in the visitSummary table and in the sourceTable_visit catalogs?  I should look at that as well.

Reply:
Yes. I don't know whether that's actually something that can occur in practice, but I could see the visitSummary table having a non-null entry for a detector, but the sourceTable having nothing.